cloudant-client/src/main/java/com/cloudant/client/api/views/AllDocsResponse.java

```diff
@@ -93,4 +93,9 @@
      */
     Map<String, String> getErrors();
 
+    /**
+     * @return the total number of rows returned by the _all_docs request.
```

Comment:
As per https://github.com/cloudant/java-cloudant/blob/bdd6d97ab64956b30a939a17eb9359a7734ec8b8/cloudant-client/src/main/java/com/cloudant/client/api/views/ViewResponse.java#L204

>the total number of rows in the view, before any filtering has taken place

Basically `total_rows` is the number of rows that are indexed by the view, not the number of rows returned in the response.

So for `_all_docs` `total_rows` is basically equivalent to the doc count of the database, which you can already get from `db.info().getDocCount()`.

For example for a db with 12 docs if I do _all_docs?key="mydoc" the response has `total_rows: 12` **not** `total_rows: 1` even though only "mydoc" will be in the response _i.e._
`db.getAllDocsRequestBuilder().keys("mydoc").build().getResponse().getDocs().size()` -> `1`
and:
`db.getAllDocsRequestBuilder().keys("mydoc").build().getResponse().getTotalRowCount()` -> `12`

Reply:
Hi @ricellis 

I would like to avoid making an additional API call `db.info().getDocCount()` to get this data and the data is already available so it's convenient to include it in the `AllDocsResponse` interface. It also aligns the Java API with the underlying REST API.

Reply:
that's fine :)
`total_rows` can be  a misleading concept so I wanted to be sure you knew what you were actually going to get from that value.

And on that note we'll need this comment to be updated so as to not mislead other users on what this returns. I'd be happy for it to say the same thing as the view i.e. `the total number of rows in the view, before any filtering has taken place`, but something more all_docs specific would also be OK if you can think of something better.

Reply:
Updated the comment to be `the total number of rows in the _all_docs view, before any filtering has taken place`

Reply:
@ricellis once this PR get's merged how soon will a new release be created and available for use?